test/externaldata/dummy-provider/scripts/generate-tls-certificate.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/../../../..
+cd "${REPO_ROOT}" || exit 1
+
+generate() {
+    # generate CA key and certificate
+    echo "Generating CA key and certificate for dummy provider..."
+    openssl genrsa -out ca.key 2048
+    openssl req -new -x509 -days 1 -key ca.key -subj "/O=Gatekeeper/CN=Gatekeeper Root CA" -out ca.crt
+
+    # generate server key and certificate
+    echo "Generating server key and certificate for dummy provider..."
+    openssl genrsa -out server.key 2048
+    openssl req -newkey rsa:2048 -nodes -keyout server.key -subj "/CN=dummy-provider.gatekeeper-system" -out server.csr
```

Comment:
is this for CI purposes only? will hardcoding to `gatekeeper-system` be an issue if it's not deployed there?

Reply:
it's for CI purposes only.

> will hardcoding to gatekeeper-system be an issue if it's not deployed there?

Yes, it will be a "bad certificate` error since the CN and the service's FQDN are no the same.

Reply:
sounds like this is for mTLS. Can we call out that this is not required for non-mTLS in the docs?